### DISCUSSION TITLE
item: requirement reading enchantment data.

### DISCUSSION THREAD

**Discussion by Drafyr** [Zrips/CMI#4369] [Feature Requests]
I'm not sure if this is already posted, I did a search and got nothing. I'd love if you could add identifiers to the item: check in specialized commands.

---

allow item: specialized command to use a custom item like if it has a certain enchantment, etc.

Can go more in-depth with command utilization.

**Comment by CloudeLecaw** [comment]
you can check for all these placeholder to be a special value
https://www.zrips.net/cmi/placeholders/

**Comment by Drafyr** [comment]
Can you point a little closer to what you're saying would draw on the items enchants as a requirement of the type of item?

**Comment by CloudeLecaw** [comment]
Just an example
you asked for check
a check can be true or false depending on your define

Example for namecheck of player
`- check:%cmi_user_name%==Cloude_Lecaw! cmi say Cloude_Lecaw Hello my Friend`

This would be execute the command only if playername is Cloude_Lecaw

Example for itemname
`- check:%cmi_iteminhand_displayname%==Multimegasword! cmi ban [playerName]
This would ban a player if he hold an item in hand with name Multimegasword

You asked for checking for Lores and enchantments. therefor you need the placeholder
%cmi_iteminhand_itemdata%

and here some description of the condition of checks from wiki
`If command start with check:[value1][==|>|>=|<|<=|!=][value2][?][#]! then we will only proceed if condition is correct. So for example`

What you cant do: Checking for all enchantments a player has
but you can check if he has no slots equiped
`f command starts with ifempty:[hand/offhand/quickbar/armor/inv/subinv]! then command will get performed if players inventory for defined area will be empty. Subinv are 27 slots, aka 3 lines of slots when opening inventory. When using inv as type, then every item in players inventory will be checked. Extra value can be defined for quickbar/maininv/subinv types, in example maininv-5 will require you to have 5 free slots in your main inventory, while subinv-10 will require to have 10 free slots in sub inventory and quickbar-3 will require you to have 3 free slots in quick bar. Example of usage: ifempty:maininv-3?! `

**Comment by Drafyr** [comment]
Thanks a bunch! :D 👍

**Comment by CloudeLecaw** [comment]
for checking for not wanted gear i would work with
hasitem check with amount 1

`If command starts with hasitem:[item:data](-amount)! then command will get performed if player has enough defined items. This will not take them on command execution.`

**Comment by Drafyr** [comment]
Yeah, I need it to do this, and take the items if it's true...

The item required is enchanted. It has a blacklisted name too. Trying to make it "deposit" into bank by taking item and paying player.

**Comment by CloudeLecaw** [comment]
You musst be a little Genius or havr the the bank plugin that allows deposit of itenms of ivrntory to bankaccount and right slot of it, with command only one command. 😜

But to be honest it's easier to simply deny the entering until the item is not in inventory anymore. Let it to the player to Store 'deposit' the item onnbank ir where else before he can Start.

**Comment by Drafyr** [comment]
I have a bank built. It has holo's for deposit/withdraw. The issue I'm having is trying to take a named item as a cost, after blacklisting the item name of course. /search didn't help. I can't figure it out. Please help? :<

**Comment by CloudeLecaw** [comment]
The first question is: "When the named item is checked when does te player should get removed the item and get money for it"?
a step by step description would be nice here, so we can see if we can help with cmi or not
